### DISCUSSION TITLE
How can Response not set cookies in responses?

### DISCUSSION THREAD

**Discussion by zwhitchcox** [mswjs/msw#2328] [Q&A] [answered]
From the docs:

>the HttpResponse class enables certain features, like mocking response cookies, which are not available when using the standard Response class

I'm confused about how you supposedly can't add cookies to a response.

```ts
new Response('hello world', {
  headers: {
    'Set-Cookie': 'x=1'
  }
})
```

Or is this just talking about in the browser?

**Comment by kettanaito** [comment, marked as answer]
Hi, @zwhitchcox. That comment references the forbidden cookie names as per the Fetch API spec, which includes the `Set-Cookie`. That limitation applies everywhere you have a faithful Fetch API implementation. Server-side implementations ignore certain aspects of the spec so you can set the cookie header in Node.js. 

MSW provides a universal `HttpResponse` class that allows for the `Set-Cookie` header anywhere.
